packages/nimbus/src/components/date-range-picker/date-range-picker.dev.mdx

```diff
@@ -0,0 +1,407 @@
+  </Stack>
+)
+```
+
+The `solid` variant (default) provides clear visual boundaries. The `ghost` variant offers a minimal appearance suitable for embedded contexts.
+
+### Granularity options
+
+The `granularity` prop controls both the level of date precision and the component's behavior:
+
+```jsx-live-dev
+const App = () => (
+  <Stack direction="column" gap="400">
+    <DateRangePicker granularity="day" />
+    <DateRangePicker granularity="minute" />
+  </Stack>
+)
+```
+
+**Behavioral differences:**
+- `granularity="day"` (default): Calendar closes automatically after selecting the end date. Time inputs are hidden.
+- `granularity="minute"` or other time values: Calendar remains open after date selection, displaying time input fields in the footer. This allows users to set both dates and times in one interaction.
+
+Available granularity options: `"day"`, `"hour"`, `"minute"`, `"second"`.
+
+### Controlled state
+
+For scenarios requiring programmatic control or coordination with other components, use controlled mode:
+
+```jsx-live-dev
+const App = () => {
+  const [value, setValue] = useState<DateRangePickerProps["value"]>(null);
+
+  return (
+    <Stack direction="column" gap="400">
+      <DateRangePicker
+        value={value}
+        onChange={setValue}
+      />
+      <Text fontSize="sm">
+        {value ? `Selected: ${value.start} to ${value.end}` : 'No selection'}
+      </Text>
+    </Stack>
+  );
+}
+```
+
+Use controlled mode when you need to:
+- Synchronize the date range with external state
+- Validate or transform date selections
+- Clear or programmatically set the date range
+
+### Time zone handling
+
+When working with `ZonedDateTime` values, configure time zone display and hour format:
+
+```jsx-live-dev
+const App = () => (
+  <Stack direction="column" gap="400">
+    <DateRangePicker
+      granularity="minute"
+      hideTimeZone={false}
+      hourCycle={24}
+    />
+    <DateRangePicker
+      granularity="minute"
+      hideTimeZone={true}
+      hourCycle={12}
+    />
+  </Stack>
+)
+```
+
+- `hideTimeZone`: Controls visibility of the time zone selector
+- `hourCycle`: Set to `12` for 12-hour format with AM/PM, or `24` for 24-hour format
+
+## Component requirements
+
+### Date value types
+
+All date values **must** use `@internationalized/date` types:
+- `CalendarDate` for date-only scenarios
+- `CalendarDateTime` for dates with times (no timezone)
+- `ZonedDateTime` for timezone-aware dates
+
+### Accessibility
+
+The DateRangePicker handles all accessibility requirements internally. However, for proper tracking and analytics, **always add an `aria-label`** if your use case warrants it:
+
+```tsx
+<DateRangePicker aria-label="Report date range" />
+```
+
+Using a `<label>` element or `aria-label` is also appropriate when the picker's purpose isn't clear from surrounding content:
+
+```tsx
+<label htmlFor="report-dates">Report date range</label>
+<DateRangePicker id="report-dates" />
+```
```

Comment:
either a label or aria-label is required, this is why in my suggestion above it says that visual labels are preferable